src/core/public/plugins/plugin.ts

```diff
@@ -90,6 +104,21 @@ export class PluginWrapper<
     return await this.instance.setup(setupContext, plugins);
   }
 
+  /**
+   * Calls `setup` function exposed by the initialized plugin.
+   * @param startContext Context that consists of various core services tailored specifically
+   * for the `start` lifecycle event.
+   * @param plugins The dictionary where the key is the dependency name and the value
+   * is the contract returned by the dependency's `start` function.
+   */
+  public async start(startContext: PluginStartContext, plugins: TPluginsStart) {
+    if (this.instance === undefined) {
+      throw new Error(`Plugin "${this.name}" can't be started since it isn't set up.`);
+    }
+
+    return await this.instance.start(startContext, plugins);
```

Comment:
`return await` is typically redundant outside of try/catch blocks, can you explain the reasoning for awaiting here?

Reply:
It's a little redundant, I agree. I think it's a bit matter of opinion here, but I've started to prefer using `await` on promise return values so that if someone does add try/catch later, they don't forget that what is being called won't throw without an `await`.